File: src/DTOs/BlogPost.cs

using System;
using System.Collections.Generic;
using System.Linq;

using ItemTTT;

namespace ItemTTT.DTOs
{
	public class BlogPost
	{
		public int?		ID				{ get; set; }
		public string	Date			{ get; set; }
		public string	ImageHtml		{ get; set; }
		public string	TitleHtmlEN		{ get; set; }
		public string	TitleHtmlFR		{ get; set; }
		public string	TitleHtmlNL		{ get; set; }
		public string	TextHtmlEN		{ get; set; }
		public string	TextHtmlFR		{ get; set; }
		public string	TextHtmlNL		{ get; set; }
		public bool		Active			{ get; set; }

		public string	Url				{ get; set; }
		public string	UrlEdit			{ get; set; }
		public string	UrlPrevious		{ get; set; }
		public string	UrlNext			{ get; set; }

		public BlogPost(Models.BlogPost src=null)
		{
			if( src == null )
			{
				Date = DateTime.Now.ToIsoDate();
				return;
			}

			ID			= src.ID;
			Date		= src.Date.ToIsoDate();
			ImageHtml	= src.ImageHtml;
			TitleHtmlEN	= src.TitleHtmlEN;
			TitleHtmlFR	= src.TitleHtmlFR;
			TitleHtmlNL	= src.TitleHtmlNL;
			TextHtmlEN	= src.TextHtmlEN;
			TextHtmlFR	= src.TextHtmlFR;
			TextHtmlNL	= src.TextHtmlNL;
			Active		= src.Active;
		}

		internal void ToModel(Models.BlogPost dst)
		{
			var date = (Date != null) ? (Date.FromIsoDate()??DateTime.Now) : DateTime.Now;

			dst.Date		= date;
			dst.ImageHtml	= ImageHtml		?? "";
			dst.TitleHtmlEN	= TitleHtmlEN	?? "";
			dst.TitleHtmlFR	= TitleHtmlFR	?? "";
			dst.TitleHtmlNL	= TitleHtmlNL	?? "";
			dst.TextHtmlEN	= TextHtmlEN	?? "";
			dst.TextHtmlFR	= TextHtmlFR	?? "";
			dst.TextHtmlNL	= TextHtmlNL	?? "";
			dst.Active		= Active;
		}
	}
}

File: app/src/DTOs/BlogPost.ts

export interface BlogPost
{
	id?			: number;
	date		: string;
	imageHtml	: string;
	titleHtmlEN	: string;
	titleHtmlFR	: string;
	titleHtmlNL	: string;
	textHtmlEN	: string;
	textHtmlFR	: string;
	textHtmlNL	: string;
	active		: boolean;
}

export class BlogPostKO
{
	public readonly	id			: KnockoutObservable<number>;
	public readonly	date		: KnockoutObservable<string>;
	public readonly	imageHtml	: KnockoutObservable<string>;
	public readonly	titleHtmlEN	: KnockoutObservable<string>;
	public readonly	titleHtmlFR	: KnockoutObservable<string>;
	public readonly	titleHtmlNL	: KnockoutObservable<string>;
	public readonly	textHtmlEN	: KnockoutObservable<string>;
	public readonly	textHtmlFR	: KnockoutObservable<string>;
	public readonly	textHtmlNL	: KnockoutObservable<string>;
	public readonly	active		: KnockoutObservable<boolean>;

	constructor(src?:BlogPost)
	{
		const self = this;

		this.id				= ko.observable( null );
		this.date			= ko.observable( '' );
		this.imageHtml		= ko.observable( '' );
		this.titleHtmlEN	= ko.observable( '' );
		this.titleHtmlFR	= ko.observable( '' );
		this.titleHtmlNL	= ko.observable( '' );
		this.textHtmlEN		= ko.observable( '' );
		this.textHtmlFR		= ko.observable( '' );
		this.textHtmlNL		= ko.observable( '' );
		this.active			= ko.observable( false );

		if( src != null )
			self.fromDTO( src );
	}

	public fromDTO(src:BlogPost) : void
	{
		const self = this;

		self.id				( src.id );
		self.date			( src.date );
		self.imageHtml		( src.imageHtml );
		self.titleHtmlEN	( src.titleHtmlEN );
		self.titleHtmlFR	( src.titleHtmlFR );
		self.titleHtmlNL	( src.titleHtmlNL );
		self.textHtmlEN		( src.textHtmlEN );
		self.textHtmlFR		( src.textHtmlFR );
		self.textHtmlNL		( src.textHtmlNL );
		self.active			( src.active );
	}

	public toDTO() : BlogPost
	{
		const self = this;
		const dst : BlogPost = <any>{};

		dst.id			= self.id();
		dst.date		= self.date();
		dst.imageHtml	= self.imageHtml();
		dst.titleHtmlEN	= self.titleHtmlEN();
		dst.titleHtmlFR	= self.titleHtmlFR();
		dst.titleHtmlNL	= self.titleHtmlNL();
		dst.textHtmlEN	= self.textHtmlEN();
		dst.textHtmlFR	= self.textHtmlFR();
		dst.textHtmlNL	= self.textHtmlNL();
		dst.active		= self.active();

		return dst;
	}
}
